lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlockBlobFileSystemE2E.cs

```diff
@@ -138,6 +139,54 @@ public void TestDeleteE2E()
             Assert.False(CheckBlobExists(blob));
         }
 
+        [Fact(Skip = SkipMessage)]
+        public void TestGetChildBlobsInContainerE2E()
+        {
+            // setup
+            string[] parentFileNames = new string[] { "sample1", "sample2", "sample3" };
+            string[] folderfileNames = new string[] { "folder1/sample4", "folder1/sample5" };
+            string[] parentListNames = new string[] { "sample1", "sample2", "sample3", "folder1/" };
+
+            foreach (string uploadedBlobName in parentFileNames.Concat(folderfileNames))
+            {
+                CloudBlockBlob blob = _container.GetBlockBlobReference(uploadedBlobName);
+                UploadFromString(blob, "hello");
+            }
+
+            // List files in the root level in container
+            ValidateChildenWithBlobs(_container.Uri, parentListNames);
+
+            // List files only in the sub-folder in the container
+            Uri folderUri = _container.GetDirectoryReference("folder1").Uri;
+            ValidateChildenWithBlobs(folderUri, folderfileNames);
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestGetChildContainerInStorageAccountE2E()
+        {
+            // List containers in the storage account
+            Uri rootUri = _fileSystem.CreateUriForPath(string.Empty);
+            ValidateChildrenWithContainers(rootUri, new string[] { _container.Name });
+        }
+
+        public void ValidateChildenWithBlobs(Uri storageBlobUri, IEnumerable<string> expectedChildBlobNames)
+        {
+            IEnumerable<Uri> blobs = _fileSystem.GetChildren(storageBlobUri).ToList();
+            IEnumerable<string> blobNames = blobs.Select(b => b.LocalPath.Replace("/" + _container.Name + "/", string.Empty));
+
+            Assert.True(expectedChildBlobNames.All(blobName => blobNames.Contains(blobName)), "blobNames has elements that are not in expectedChildBlobNames");
+            Assert.Equal(expectedChildBlobNames.Count(), blobNames.Count());
+        }
+
+        public void ValidateChildrenWithContainers(Uri rootUri, IEnumerable<string> expectedContainerNames)
```

Comment:
No reason for another method. This method is basically identical to ValidateChildenWithBlobs.

Reply:
Hmm .. they are different .. in how we extract out the blob names or container names as the case may be from the output uri. I could try to consolidate it even more but I am afraid that we lose readability (containers vs blobs). Lets chat offline if you are not convinced with the reasoning.